src/Screens/Main/Chat/Title.tsx

```diff
@@ -1,49 +1,100 @@
 import React from 'react';
 import RN from 'react-native';
 import { SafeAreaView } from 'react-navigation';
+
 import * as redux from 'redux';
 import * as ReactRedux from 'react-redux';
-
 import * as state from '../../../state';
 import * as selectors from '../../../state/selectors';
 import * as actions from '../../../state/actions';
 
+import useLayout from 'src/lib/use-layout';
+
 import colors from '../../components/colors';
 import { cardBorderRadius } from '../../components/Card';
 import fonts from '../../components/fonts';
 import getBuddyColor from '../../components/getBuddyColor';
-import DropDown, { DropDownItem } from 'src/Screens/components/DropDownMenu';
-import { Dialog } from 'src/Screens/components/Dialog';
+
+import DropDown, { DropDownItem } from '../../components/DropDownMenu';
+import { Dialog } from '../../components/Dialog';
+import { MessageId } from '../../../localization';
 
 type StateProps = {
   name: string;
 };
+
 type DispatchProps = {};
+
 type OwnProps = {
   style?: RN.StyleProp<RN.ViewStyle>;
   onPress: () => void | undefined;
   buddyId: string;
 };
+
 type Props = OwnProps & DispatchProps & StateProps;
 
+type DialogProperties = {
+  text: MessageId;
+  button: MessageId;
+  onPress: () => void | undefined;
+  type: 'warning' | undefined;
+};
+
 const Title: React.FC<Props> = ({ style, onPress, name, buddyId }) => {
-  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [isDropdownOpen, setDropdownOpen] = React.useState(false);
+
+  const [isDialogOpen, setDialogOpen] = React.useState(false);
+
+  const [{ height }, onLayout] = useLayout();
+
   const color = getBuddyColor(buddyId);
+
   const isBanned = ReactRedux.useSelector(selectors.getIsBanned(buddyId));
+
   const dispatch = ReactRedux.useDispatch<redux.Dispatch<actions.Action>>();
-  const banBuddy = () => {
-    dispatch({ type: 'buddies/ban/start', payload: { buddyId } });
+
+  const banBuddy = (status: 'Active' | 'Banned') => {
+    dispatch({
+      type: 'buddies/changeStatus/start',
+      payload: { buddyId, status },
+    });
   };
-  const handleBan = () => {
-    banBuddy();
+
+  const closeDropdownAndDialog = () => {
+    setDropdownOpen(false);
+    setDialogOpen(false);
+  };
+
+  const handleBan = (status: 'Active' | 'Banned') => {
```

Comment:
"Unban" | "Banned"? Naming is hard.

Reply:
Now have type BanActions which is = 'Ban' | 'Unban'